### ISSUE TITLE
Is there any plan to migrate this framework to Angular 2.0 ?

### ISSUE THREAD

**Issue by sharathbangera6** [angular-dashboard-framework/angular-dashboard-framework#214]

**Comment by raulgomis** [comment]
+1

**Comment by raulgomis** [comment]
I really think that this project should be migrated to Angular 2.0, now that Angular2 is official. We will wait for @sdorra opinion regarding the migration, but I think both versions should be maintained during some time:
- adf 0.x could be kept in Angular 1.x during some time (bugfixes and minor features).
- adf 1.x could be released in Angular 2.x.

What do you think? Any volunteers to help with the migration?

**Comment by sdorra** [comment]
I've started some research and tests with angular 2. I would love to implement an adf version for angular 2, but it needs some rethinking. My current thoughts:

* a widget is represented by a component, which extends a specific widget class
* widgets should live in a separate modules
* the type of the widget should be stored in the model
* function such as edit or reload could be implemented by extending a specific interface and implementing the methods
* the model should be defined with interfaces, which can be cast from json

But  i have not much experience with angular 2 and so i have to figure out some things e.g.: 

* how do i embed a dynamic component into another component?
* how to load all registered widgets on application start?

And there will be much more questions during the development.

**Comment by ivan-saorin** [comment]
Hi Sebastian, another possibility is of course Aurelia. I tried it out and it is much better than angular 2 (no troll war intended, It's just my opinion).

At the and of the day Angular 2 is way less compelling than the 1.x branch as It miss the huge amount of ready made objects.

**Comment by sdorra** [comment]
Hi folks a started a proof of concept at [adf2-poc](https://github.com/angular-dashboard-framework/adf2-poc).

**Comment by meikeric** [comment]
@sdorra I created a PR to update the packages on your POC.  It would not run with a straight clone. PR: https://github.com/angular-dashboard-framework/adf2-poc/pull/2

**Comment by jayhamilton** [comment]
I actually have an angular 2.x version that was somewhat inspired by this project. At some point when I polish it off a bit more I will attempt to post it.

**Comment by pietromarrone** [comment]
We have to create a project on top of the angular dashboard framework, but it has to be based on angular2. 

We can fund the community or developer capable of migrating adf to angular2.

Requirements are those of adf, with excellent code quality and as modular as possible.

Write in private for more details and to define time and money. 

P.S. the code must be released on github

**Comment by jayhamilton** [comment]
@pietromarrone did you take a look at my project? It is based on Angular 4. The most significant difference between Angular 2 and Angular 4, from my project's perspective, is the Angular animation library.  And that  difference is simply in the change in location where Angular provides the library.

**Comment by pietromarrone** [comment]
Yes I did. The problem is not Angular4, actually I wrote Angular2 but last Angular is even better.
But it didn't start, and as you wrote on the reademe we cannot open issue.
So, in the end, if you could make it running, so I can evaluate what is missing, the code quality, and try to make an agreement.
Regards

**Comment by jayhamilton** [comment]
@pietromarrone There is no reason it should not start. You need to make sure you have cli version 1.3 installed. ng serve should work.

**Comment by pmink** [comment]
@jayhamilton it wouldn't work for me either initially. Seems fine after I updated the Angular Material references.

**Comment by jayhamilton** [comment]
@pmink I have not updated to the latest version of Angular Material yet. This will be the very next thing that I will do.

**Comment by meikeric** [comment]
@pmink Take a look at https://github.com/AngularDashboardFramework/AngularDashboardFramework
This is an evolution of @jayhamilton’s project I’ve been working on.  It’s broken out into an Angular library and an example CLI project.  I’ll have a new release in a day or two.
